Column Reordering in Admin "Basic" Tab

shaahmad opened the pull request:
![ezgif-1-4c85927366b2](https://user-images.githubusercontent.com/11683872/101555922-a0701400-3987-11eb-8bf6-6830ecb4d46e.gif)

MarcelGerber commented:
Great, I like it!

Maybe it would be more obvious that dimensions can be dragged if we include an arrow as drag handle, like we do in the "Data" tab? (can also do this later on)
![image](https://user-images.githubusercontent.com/2641501/101616649-ee305f00-3a0f-11eb-810a-83be420410ed.png)

Also, all these different variables (`dimensionsOrderedAsInPersistedSelection`, `dimensionsOrderedAsDisplayed`, `legacySelectionOrderedAsDisplayed`) are quite involved, can we maybe trim them down in some way? If not, then that's also alright with me.

shaahmad commented:
@MarcelGerber I added the arrows and fixed the layout too. Thought hard about renaming the functions/variables, but couldn't think of a way yet that would simplify without reducing the amount of information in the name. I'll merge as-is and keep thinking about it.
